Reverse ETL source primary key setting (#909)
rudderlabs/rudderstack-docs: docs/sources/reverse-etl/common-settings/importing-data-using-tables.mdx

 Detailed technical description on the various configuration settings to sync data from your warehouse tables.
---

<div class="infoBlock">
This guide is applicable for the Reverse ETL sources configured using the <strong>Table</strong> option. If you have configured your Reverse ETL source using the <strong>Model</strong> option, refer to the <Link to="/sources/reverse-etl/common-settings/importing-data-using-models/">Importing Data using Models</Link> guide.
</div>

> RudderStack gives you two options to map your columns to specific destination fields while importing the data:

Reviewer: ```suggestion
RudderStack gives you two options to map your warehouse columns to specific destination fields while importing the data:
```

> - Map with <Link to="/sources/reverse-etl/features/visual-data-mapper/">Visualizer</Link> (Currently supported only for selective destinations)

Reviewer: ```suggestion
- Map with Visualizer (Refer to the <Link to="/sources/reverse-etl/features/visual-data-mapper/">Visualizer</Link> guide which mentions the supported destinations.)
```

- Map with JSON

This guide lists the JSON mapping settings required to import and sync data from your warehouse table to the specified destination.

## Data import settings

The settings to import and sync data from your warehouse table are as shown:

<img src="../../../assets/warehouse-actions-sources/data-import-settings-2.png" alt="Data import settings in RudderStack" />

- **Schema**: Specify your warehouse schema.
- **Table**: Specify the table residing in the above schema from which you want RudderStack to import the data.
- **Sync mode**: Select the sync mode that RudderStack uses to sync your data.

<div class="infoBlock">
RudderStack supports two modes for syncing your data - <strong>Upsert</strong> and <strong>Mirror</strong>. For more information on these modes, refer to the <Link to="/sources/reverse-etl/common-settings/sync-modes/">Sync Modes</Link> guide.
</div>

> - **Primary Key**: Select the column to uniquely identify your user records.

Reviewer 2: ```suggestion
- **Primary Key**: Select the column to uniquely identify your records in warehouse.
```

Also a hint like, syncs use this primary key for incremental syncs

> - **Primary Key**: Select the column to uniquely identify your records in the warehouse.

Reviewer: ```suggestion
- **Primary Key**: Select a column from the table specified above to uniquely identify your records in the warehouse.
```
